proto/messages.go

```diff
@@ -307,7 +306,7 @@ func (w *slicewriter) Slice() []byte {
 // off part of the last message. This also means that the last message can be
 // shorter than the header is saying. In such case just ignore the last
 // malformed message from the set and returned earlier data.
-func readMessageSet(r io.Reader, size int32) ([]*Message, error) {
+func readMessageSet(r io.Reader, size int32, version int16) ([]*Message, error) {
```

Comment:
Add a comment above about the new `version` parameter.

Reply:
done